crates/forge_app/src/app.rs

```diff
@@ -99,12 +100,43 @@ impl<S: Services> ForgeApp<S> {
             chat.event = chat.event.attachments(attachments);
         }
 
+        let mut paths = Vec::new();
+        let agent_md = environment.base_path.join("AGENTS.md");
+        if !paths.contains(&agent_md) {
+            paths.push(agent_md);
+        }
+        if let Some(git_root_path) = get_git_root(&environment.cwd).await {
+            let git_agent_md = git_root_path.join("AGENTS.md");
+            if !paths.contains(&git_agent_md) {
+                paths.push(git_agent_md);
+            }
+        }
+        let cwd_agent_md = environment.cwd.join("AGENTS.md");
+        if !paths.contains(&cwd_agent_md) {
+            paths.push(cwd_agent_md);
+        }
```

Comment:
Create a separate service that gets all the agents.md file: `CustomInstructionsSerivce`. The service should figure out how to get all the agents.md files, read them and cache them. App.rs should simply call that service to get that data.